src/datasets/queries/categorical.query.ts

```diff
@@ -0,0 +1,21 @@
+import { CategoricalPoint } from '../metas/types';
+import { DAY } from '../../utils/timeUnits';
+import { DataCube } from '../../models/data-cube/data-cube.model';
+import { ResultRow } from '../../models/data-cube/types';
+import { unique } from '../../utils/misc';
+import { SummaryGroup } from '../summarizations/types';
+import { XYDatum } from '../metas/types';
+
+export type CategoricalQueryOptions = {};
+
+export interface CategoricalDatum<S> extends XYDatum<CategoricalPoint> {
```

Comment:
No they are not the same. `TimeSeriesDatum` takes `XYPoint<string, number>` as point type, but `CategoricalDatum` takes `XYPoint<string, number>`. Separating them also help for future development since they should not be the same for the nature of difference in bar/pie chart and line chart.